app/scripts/components/exploration/index.tsx

```diff
@@ -69,11 +75,91 @@ const tourProviderStyles = {
 };
 
 function Exploration() {
+  const [projection, setProjection] = useState(projectionDefault);
+
+  const {
+    mapBasemapId,
+    setBasemapId,
+    labelsOption,
+    boundariesOption,
+    onOptionChange
+  } = useBasemap();
+
   const datasets = useAtomValue(timelineDatasetsAtom);
+  const setDatasets = useSetAtom(timelineDatasetsAtom);
+
+  useReconcileWithStacMetadata(datasets, setDatasets);
```

Comment:
From this line to line 158, some of the features can't be overridden (or can't be optional) to use Exploration properly. ex. `onStyleUpdate`  attaches TileURLs as metadata to datasets. If this function gets overridden by a user-passed function, `LoadIntoGIS` functionality from the timeline can't work anymore. (Attaching a screenshot of LoadIntoGIS functionality). Same with reconciling StacMetadata. In my mind,  this is one of the crucial features of ExplorationMap that can be done out-of-box.

![Screen Shot 2024-05-21 at 2 57 35 PM](https://github.com/NASA-IMPACT/veda-ui/assets/4583806/65997a86-b010-42e5-8e42-31426066d88c)

Now that I can see more clearly that Exploration is a tricky component. There are global states that exist on this page that only make sense inside of the container together (Timeline and Map), so I am not sure if each component can be used outside of its wrapper. How do you imagine users will construct the exploration page? (I think it would be ok if we ask users to use exploration as it is like`<Exploration datasets={datasets}>` at least for the short-term future? but I might be missing something - !  lmk if you want to discuss further etc. )

Reply:
@hanbyul-here @sandrahoang686 Thanks, lots of good pointers here!

> How do you imagine users will construct the exploration page?

Not all is done in the PR, but I imagined few things on how the Exploration page could evolve:

1. Allow more functionality overriding
2. Allow hiding / showing of certain controls and elements within the ExplorationMap and the Timeline (not necessarily done in this PR)

Reading through the PR comments, point 1 above seems that is easier said than done, and I don't have a good answer here. 

> I think it would be ok if we ask users to use exploration as it is like<Exploration datasets={datasets}> at least for the short-term future

It seems like this is the way to go, or until the feature prompts for more flexibility. But I can't imagine a scenario where I would like to only use the timeline widget on a page, or where I would only like to use the Exploration map on a page. They really seem like one logical core feature, both in design and functionality.

Reply:
I think it makes sense to couple both the map and timeline as a core feature as the way forward. Lets not worry too much about functionality overriding till we meet those usecases requiring it, we can build to be flexible though where we see current usecases may need it 👍🏼

Reply:
An additional two cents here is we might want to see the composability of more broken-down components, not the overridability (is this a word) of already composed components. In my mind, Exploration map is already a composed component that needs to serve a specific purpose; I think the goal of this component doesn't have to include the flexibility of how it can be used by users. (To find an example, disabling AOI control doesn't make sense in the Exploration map because it is directly related to the analysis feature)

Reply:
Okay, I think all of the comments should be addressed now